### ISSUE TITLE
Option to set CPU type rather than backend (SSE, AVX2)

### ISSUE THREAD

**Issue by pramodk** [BlueBrain/nmodl#684]
Hello @georgemitenkov,

We want to benchmark performance with SSE, AVX2, AVX-512 etc. I was working on adding CLI option to set cpu features from CLI. I realised that we have code below which didn't work as expected: 

```cpp
void LLVMBenchmark::run_benchmark(const std::shared_ptr<ast::Program>& node) {
    // Set the codegen data helper and find the kernels.
    auto codegen_data = codegen::CodegenDataHelper(node, llvm_visitor.get_instance_struct_ptr());
    std::vector<std::string> kernel_names;
    llvm_visitor.find_kernel_names(kernel_names);

    // Get feature's string and turn them off depending on the backend.
    std::vector<std::string> features = get_cpu_features();
    logger->info("Backend: {}", backend);
    if (backend == "avx2") {
        // Disable SSE.
        logger->info("Disabling features:");
        disable("sse", features);
        disable("sse2", features);
        disable("sse3", features);
        disable("sse4.1", features);
        disable("sse4.2", features);
    } else if (backend == "sse2") {
        // Disable AVX.
        logger->info("Disabling features:");
        disable("avx", features);
        disable("avx2", features);
    }

    std::string features_str = llvm::join(features.begin(), features.end(), ",");
    std::unique_ptr<llvm::Module> m = llvm_visitor.get_module();
...
```

Instead of enabling/disabling features, don't you think we should set directly CPU type instead of always getting default CPU?  i.e. like `-cpu` arg for other commands, what if we take target cpu type like `haswell`  as CLI arg and use that? This way, we won't need to enable or disable specific set of features. If we say it's haswell then it should automatically add all cpu features?

**Comment by georgemitenkov** [comment]
I was about to remove that actually)

Yes, we can add a cpu option and I think it would be useful to have it for benchmarking! Disabling features is wrong and we should not do that for sure.

Currently we have this:
```c++
// jit_driver.cpp
    std::string cpu(llvm::sys::getHostCPUName());
    std::unique_ptr<llvm::TargetMachine> tm(
        target->createTargetMachine(target_triple, cpu, features, {}, {}));
    if (!tm)
        throw std::runtime_error("Error: could not create the target machine\n");
```
So instead of querying for a cpu we can just pass the one from the command line.

**Comment by pramodk** [comment]
fixed by #686
